workout.py

```diff
@@ -74,10 +74,10 @@ def generate_exercises_for_category(self,
         cat_excercises = self.shuffled_grouped_exercises[cat.id]
         for i in count():
             exc = cat_excercises[i % len(cat_excercises)]
-            if category_time > exc.time and exc.id not in set(
-                         map(lambda exc: exc.id, blacklisted_exercises)):
-                exc_list.append(exc)
-                category_time -= exc.time
+            if category_time > exc.time:
+                if exc.id not in set(map(lambda exc: exc.id, blacklisted_exercises)):
```

Comment:
why do you need a set here?

Reply:
testing for inclusion (`not in`) only works on a set